### ISSUE TITLE
JVM throws an error when opening some pdf files

### ISSUE THREAD

**Issue by tkunse** [zaro/pdf-metadata-editor#11]
Hello,

Thanks for your great editor. It really helps me a lot.
But I have encountered a minor problem. When I open some pdf files, JVM gives me an error message:
java.io.IOException: Invalid byte 3 of 3-byte UTF-8 sequence.

An example:
http://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-57p1r3.pdf

It seems that the error occurs when reading the pdf file. Do you have any way to solve this problem?

Thanks again!

**Comment by tkunse** [comment]
Oh, I forgot to mention another minor bug.

I have set the default values, and it is expected that when I open a PDF document, leave all the text fields blank and directly click on the "save" button, the program should read the default values and fill in the blanks automatically. But the fact is that it works only after I open preferences window. I have to click on the preference button every time I open the software. So is it possible for you to make it read the settings when it opens?

Thanks!

**Comment by tkunse** [comment]
After such a long time, I finally realized what's the problem. (Because I myself have written several Java programs too and experienced several problems when trying to distribute. However, I do not upload the code to github for some security reasons.)
This error message is all caused by the encoding. Since I am using this software on Windows, the system's default encoding is not UTF-8. For this reason alone, Java programs cannot perform normally when reading something that is encoded by UTF-8, such as the metadata in PDF files.
So the solution is to add a `-Dfile.encoding=UTF-8` to the JVM option. This can solve most problems. However, when a PDF document is protected using permission passwords, exceptions are thrown again. But it is not a serious problem then, for I can just remove the passwords and use this tool.
Sorry for the disturbance (if any).

**Comment by courtiol** [comment]
How does one launch the GUI with this argument? I tried by failed...

**Comment by zaro** [comment]
@courtiol   Check this screenshot, I think it will make it clear, where you can add it .

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/6fa537df-9fcc-4ac3-be0f-ba3866372507" />

**Comment by courtiol** [comment]
Thanks, but I am running Linux and I cannot find something similar.

**Comment by zaro** [comment]
@courtiol  ok, how did you install it ? deb ? rpm? zip ?

The structure of the app is very very similar also for linux.

**Comment by zaro** [comment]
Next to the bin/ folder where the exe is you will find app/ folder that contains the *.cfg files.

________________________________

**Comment by courtiol** [comment]
Found it in `/opt/pdf-metadata-editor/lib/app/`.
Thanks

**Comment by courtiol** [comment]
Unfortunately that does not solve the import problem I have with a PDF.
See attached.

[An-8000-year-history-of-landscape-climate-and-copper-exploitation-in-the-Middle-East-the-Wadi-Faynan-and-the-Wadi-Dana-National-Reserve-in-southern-Jordan.pdf](https://github.com/user-attachments/files/25160836/An-8000-year-history-of-landscape-climate-and-copper-exploitation-in-the-Middle-East-the-Wadi-Faynan-and-the-Wadi-Dana-National-Reserve-in-southern-Jordan.pdf)

**Comment by fr0stb1rd** [comment]
with `v4.0.1 latest` (aur package)
file: `An-8000-year-history-of-landscape-climate-and-copper-exploitation-in-the-Middle-East-the-Wadi-Faynan-and-the-Wadi-Dana-National-Reserve-in-southern-Jordan.pdf`
file url: `https://github.com/user-attachments/files/25160836/An-8000-year-history-of-landscape-climate-and-copper-exploitation-in-the-Middle-East-the-Wadi-Faynan-and-the-Wadi-Dana-National-Reserve-in-southern-Jordan.pdf`

```
[2026-02-27 07:58:06] [INFO ] [pmedit.Main] - Parsed command line: CommandLine(noGui=false, batchGui=false, showHelp=false, command=, files=[])
[2026-02-27 07:58:06] [INFO ] [pmedit.Main] - publish: CommandLine(noGui=false, batchGui=false, showHelp=false, command=, files=[])
[2026-02-27 07:58:06] [INFO ] [pmedit.Main] - executeCommandSwingWorker: CommandLine(noGui=false, batchGui=false, showHelp=false, command=, files=[])
[2026-02-27 07:58:06] [INFO ] [pmedit.Main] - executeCommand fileName: null
[2026-02-27 07:58:06] [INFO ] [pmedit.Main] - open editor: null
[2026-02-27 07:58:06] [INFO ] [pmedit.ext.ExtensionLoader] - Loaded extensions in 33 ms
[2026-02-27 07:58:06] [INFO ] [pmedit.ext.ExtensionLoader] - Loaded extension pmedit.ext.ProExtension in 37 ms
[2026-02-27 07:58:13] [INFO ] [pmedit.ui.MainWindow] - Received Dropped files: [/home/xxx/Downloads/An-8000-year-history-of-landscape-climate-and-copper-exploitation-in-the-Middle-East-the-Wadi-Faynan-and-the-Wadi-Dana-National-Reserve-in-southern-Jordan.pdf]
[2026-02-27 07:58:13] [INFO ] [pmedit.Main] - publish: CommandLine(noGui=false, batchGui=false, showHelp=false, command=, files=[/home/xxx/Downloads/An-8000-year-history-of-landscape-climate-and-copper-exploitation-in-the-Middle-East-the-Wadi-Faynan-and-the-Wadi-Dana-National-Reserve-in-southern-Jordan.pdf])
[2026-02-27 07:58:13] [INFO ] [pmedit.Main] - executeCommandSwingWorker: CommandLine(noGui=false, batchGui=false, showHelp=false, command=, files=[/home/xxx/Downloads/An-8000-year-history-of-landscape-climate-and-copper-exploitation-in-the-Middle-East-the-Wadi-Faynan-and-the-Wadi-Dana-National-Reserve-in-southern-Jordan.pdf])
[2026-02-27 07:58:13] [INFO ] [pmedit.Main] - executeCommand Found empty editor, using it to load : /home/xxx/Downloads/An-8000-year-history-of-landscape-climate-and-copper-exploitation-in-the-Middle-East-the-Wadi-Faynan-and-the-Wadi-Dana-National-Reserve-in-southern-Jordan.pdf
[2026-02-27 07:58:13] [INFO ] [pmedit.ui.MainWindow] - loadFile(/home/xxx/Downloads/An-8000-year-history-of-landscape-climate-and-copper-exploitation-in-the-Middle-East-the-Wadi-Faynan-and-the-Wadi-Dana-National-Reserve-in-southern-Jordan.pdf) begin
[2026-02-27 07:58:13] [ERROR] [pmedit.ext.BasicPdfWriter] - Failed to parse XMP metadata, invalid XML
org.apache.xmpbox.xml.XmpParsingException: Failed to instantiate DateType property with value 0-01-01T00:00:00Z in xap:CreateDate
	at org.apache.xmpbox.xml.DomXmpParser.createProperty(DomXmpParser.java:373)
	at org.apache.xmpbox.xml.DomXmpParser.parseChildrenAsProperties(DomXmpParser.java:322)
	at org.apache.xmpbox.xml.DomXmpParser.parseDescriptionRoot(DomXmpParser.java:250)
	at org.apache.xmpbox.xml.DomXmpParser.parse(DomXmpParser.java:201)
	at pmedit.MetadataInfoUtils.loadXMPMetadata(MetadataInfoUtils.java:16)
	at pmedit.ext.BasicPdfReader.loadFromPDF(BasicPdfReader.java:157)
	at pmedit.ui.MainWindow.reloadFileFromDocument(MainWindow.java:478)
	at pmedit.ui.MainWindow.loadFile(MainWindow.java:447)
	at pmedit.ui.MainWindow.loadFile(MainWindow.java:430)
	at pmedit.Main.executeCommandSwingWorker(Main.java:121)
	at pmedit.Main$CommandsExecutor.process(Main.java:287)
	at java.desktop/javax.swing.SwingWorker$2.run(Unknown Source)
	at java.desktop/sun.swing.AccumulativeRunnable.run(Unknown Source)
	at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(Unknown Source)
	at java.desktop/sun.swing.AccumulativeRunnable.run(Unknown Source)
	at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(Unknown Source)
	at java.desktop/javax.swing.Timer.fireActionPerformed(Unknown Source)
	at java.desktop/javax.swing.Timer$DoPostEvent.run(Unknown Source)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: Failed to instantiate DateType property with value 0-01-01T00:00:00Z
	at org.apache.xmpbox.type.TypeMapping.instanciateSimpleProperty(TypeMapping.java:171)
	at org.apache.xmpbox.xml.DomXmpParser.manageSimpleType(DomXmpParser.java:450)
	at org.apache.xmpbox.xml.DomXmpParser.createProperty(DomXmpParser.java:360)
	... 26 common frames omitted
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source)
	at org.apache.xmpbox.type.TypeMapping.instanciateSimpleProperty(TypeMapping.java:165)
	... 28 common frames omitted
Caused by: java.lang.IllegalArgumentException: Value given is not allowed for the Date type: class java.lang.String, value: 0-01-01T00:00:00Z
	at org.apache.xmpbox.type.DateType.setValue(DateType.java:127)
	at org.apache.xmpbox.type.AbstractSimpleProperty.<init>(AbstractSimpleProperty.java:59)
	at org.apache.xmpbox.type.DateType.<init>(DateType.java:57)
	... 32 common frames omitted
[2026-02-27 07:58:13] [INFO ] [pmedit.ui.MainWindow] - loadFile(/home/xxx/Downloads/An-8000-year-history-of-landscape-climate-and-copper-exploitation-in-the-Middle-East-the-Wadi-Faynan-and-the-Wadi-Dana-National-Reserve-in-southern-Jordan.pdf) success
[2026-02-27 07:58:27] [INFO ] [pmedit.Main] - Received windowClosing for pmedit.ui.MainWindow[frame0,397,114,1157x875,invalid,layout=java.awt.BorderLayout,title=Pdf Metadata Editor,resizable,normal,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,0,0,1157x875,layout=com.formdev.flatlaf.ui.FlatRootPaneUI$FlatRootLayout,alignmentX=0.0,alignmentY=0.0,border=com.formdev.flatlaf.ui.FlatRootPaneUI$FlatWindowBorder@65ef8293,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
[2026-02-27 07:58:27] [INFO ] [pmedit.Main] - maybeExit() batchInstances=0, editorInstances=0, cmdQueue=0
[2026-02-27 07:58:27] [INFO ] [pmedit.Main] - No instances left, exiting...
```

**Comment by zaro** [comment]
@fr0stb1rd  1st I want to point out that the metadata in this file is indeed invalid. 

prior to v4.0.1 , the file would simply fail to open. in 4.0.1 the file will open, with empty XMP metadata.

So, I am not sure what do you expect, maybe you want to elaborate ?
BTW making the editor  work with every possible invalid XML, is not a priority for me at the moment.

**Comment by fr0stb1rd** [comment]
@zaro I completely agree that trying to fix every possible invalid XML/XMP variation shouldn't be a priority, and opening the file with empty metadata is definitely better than crashing.

My intention in sharing the logs was that silently failing in the background might confuse end-users. They might just see an empty metadata screen and wonder if the app is broken or if the file actually has no data.

I think it would be helpful to show a brief info/warning message via the UI (e.g., a simple warning dialog or a status bar notice) stating that the metadata couldn't be fully parsed due to invalid formatting.

If you're open to this approach, I'd be happy to implement the warning mechanism and submit a PR. Let me know what you think.

**Comment by zaro** [comment]
@fr0stb1rd  sure. It's a good idea.

Maybe open a minimal PR, and we can discuss further there.

**Comment by zaro** [comment]
@fr0stb1rd Just pushed v4.1.0beta1. It has a new warning/error indicator with a log of errors while loading the metadata. It still very minimal, but it's a start.

**Comment by fr0stb1rd** [comment]
@zaro thank you for update. i tried it. it works.

**Comment by zaro** [comment]
@fr0stb1rd  v4.1.0 is out. Closing this issue.
